### PR TITLE
Added __LINE__ const

### PR DESCRIPTION
**Pull request by CrazyHackGUT**
Added `__line` const. He always equals current code line.

Test code:
```sourcepawn
#include <sourcemod>

public void OnPluginStart()
{
    PrintToServer("Check magic Pawn consts...");

    PrintToServer("cellbits = %d", cellbits);
    PrintToServer("cellmax  = %d", cellmax);
    PrintToServer("cellmin  = %d", cellmin);

    PrintToServer("charbits = %d", charbits);
    PrintToServer("charmax  = %d", charmax);
    PrintToServer("charmin  = %d", charmin);

    PrintToServer("debug    = %d", debug);

    PrintToServer("__Pawn   = %x", __Pawn);
    PrintToServer("__line   = %d", __line);

    PrintToServer("ucharmax = %d", ucharmax);
}
```

example running test code
line number

### CODE REVIEW

**Comment by Drifter321** [comment]
This should probably be __LINE__ to be the same as the amxmodx one since it matches other languages. https://github.com/alliedmodders/amxmodx/blob/1a56577f7a3c8167f34cec0bbd443402f5636dcf/compiler/libpc300/sc1.c#L1463

File: compiler/sc2.cpp [hunk 1]
```diff
@@ -388,6 +389,9 @@ static void readline(unsigned char *line)
       line+=strlen((char*)line);
     } /* if */
     fline+=1;
+    sym=findconst("__line",NULL);
+    assert(sym!=NULL);
+    sym->addr_=fline;
```
Comment (rsKliPPy) [inline compiler/sc2.cpp]: You should probably do `sym->setAddr(fline);` instead to be consistent.

Comment (CrazyHackGUT) [reply compiler/sc2.cpp]: Thanks.

**Comment by dvander** [comment]
this looks fine but it deserves additions to compiler/tests - ideally two, one testing that __LINE__ works as normal and another that it works in macro expansions.

**Comment by CrazyHackGUT** [comment]
Thanks. I added tests.

**Comment by dvander** [comment]
Thanks!

**Pull request merged**